nixos/modules/services/networking/netbird.nix

```diff
@@ -1,171 +1,336 @@
+          port = 51820;
+          name = "netbird";
+          interface = "wt0";
+        };
+        ```
+      '';
+    };
     package = mkPackageOption pkgs "netbird" { };
 
-    tunnels = mkOption {
+    clients = mkOption {
       type = attrsOf (
         submodule (
           { name, config, ... }:
           {
             options = {
               port = mkOption {
                 type = port;
-                default = 51820;
+                example = literalExpression "51820";
                 description = ''
-                  Port for the ${name} netbird interface.
+                  Port the Netbird client listens on.
+                '';
+              };
+
+              name = mkOption {
+                type = str;
+                default = "netbird-${name}";
+                description = ''
+                  Primary name for given client, used in naming of directories, services and users.
+                '';
+              };
+
+              interface = mkOption {
+                type = str;
+                default = "netbird-${name}";
+                description = ''
+                  Name of the network interface managed by this client.
                 '';
               };
 
               environment = mkOption {
                 type = attrsOf str;
                 defaultText = literalExpression ''
                   {
-                    NB_CONFIG = "/var/lib/''${stateDir}/config.json";
-                    NB_LOG_FILE = "console";
-                    NB_WIREGUARD_PORT = builtins.toString port;
-                    NB_INTERFACE_NAME = name;
-                    NB_DAMEON_ADDR = "/var/run/''${stateDir}"
+                    NB_CONFIG = "/var/lib/''${config.name}/config.json";
+                    NB_DAEMON_ADDR = "unix:///var/run/''${config.name}/sock";
+                    NB_INTERFACE_NAME = config.interface;
+                    NB_LOG_FILE = mkOptionDefault "console";
+                    NB_LOG_LEVEL = config.logLevel;
+                    NB_WIREGUARD_PORT = toString config.port;
                   }
                 '';
                 description = ''
                   Environment for the netbird service, used to pass configuration options.
                 '';
               };
 
-              stateDir = mkOption {
-                type = str;
-                default = "netbird-${name}";
+              autoStart = mkOption {
+                type = bool;
+                default = true;
+                description = ''
+                  Start the service with the system.
+
+                  As of 2024-02-13 it is not possible to start a Netbird client daemon without immediately
+                  connecting to the network, but it is [planned for a near future](https://github.com/netbirdio/netbird/projects/2#card-91718018).
+                '';
+              };
+
+              openFirewall = mkOption {
+                type = bool;
+                default = true;
```

Comment:
I would set the default value to `false`, the opening of a port in the firewall should be a conscious decision

Reply:
I'd argue it doesn't make much sense setting it up to connect to LAN peers over the internet by default while the port is securely authenticated anyway.

There is some precedence for secure connection services:
- https://github.com/NixOS/nixpkgs/blob/f9d39fb9aff0efee4a3d5f4a6d7c17701d38a1d8/nixos/modules/services/networking/wgautomesh.nix#L51
- https://github.com/NixOS/nixpkgs/blob/f9d39fb9aff0efee4a3d5f4a6d7c17701d38a1d8/nixos/modules/services/networking/ssh/sshd.nix#L220
- https://github.com/NixOS/nixpkgs/blob/f9d39fb9aff0efee4a3d5f4a6d7c17701d38a1d8/nixos/modules/programs/mosh.nix#L13